domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterSingleton.kt

```diff
@@ -0,0 +1,31 @@
+package org.oppia.android.domain.platformparameter
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+import org.oppia.android.util.platformparameter.PlatformParameter
+
+/** Singleton class which helps in storing and providing Platform Parameters at runtime. */
+@Singleton
+class PlatformParameterSingleton @Inject constructor() {
+  var platformParameterMap: Map<String, ParameterValue> = mapOf()
+
+  /** Initialize platformParameterMap if it is not yet initialised. */
```

Comment:
Sure will do that

Reply:
Ah got it, is there a reason we are writing our own setter then?  If this is meant to initialize the map only once then I recommend adding error handling if the map has already been initialized. Otherwise why not use the existing setter?

Reply:
Yeah we can use the default setter also. I will find about how to implement the same functionality with it

Reply:
I see this has been renamed, were there collisions with the setter?

Reply:
As the property is now private we need a setter to access it